src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -788,4 +746,90 @@ DictionaryArray CameraFactory::get_input_metadata()
     return metadata;
 }
 
+void CameraFactory::add_film_metadata(DictionaryArray& metadata)
+{
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "film_dimensions")
+            .insert("label", "Film Dimensions")
+            .insert("type", "text")
+            .insert("use", "required"));
+
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "film_width")
+            .insert("label", "Film Width")
+            .insert("type", "text")
+            .insert("use", "required"));
+
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "film_height")
+            .insert("label", "Film Height")
+            .insert("type", "text")
+            .insert("use", "required"));
+
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "aspect_ratio")
+            .insert("label", "Aspect Ratio")
+            .insert("type", "text")
+                .insert("use", "required"));
+}
+
+void CameraFactory::add_lens_metadata(DictionaryArray& metadata)
+{
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "focal_length")
+            .insert("label", "Focal Length")
+            .insert("type", "text")
+            .insert("use", "required"));
+
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "horizontal_fov")
+            .insert("label", "Horizontal FOV")
+            .insert("type", "numeric")
+            .insert("min",
+                Dictionary()
+                    .insert("value", "1.0")
+                    .insert("type", "soft"))
+            .insert("max",
+                Dictionary()
+                    .insert("value", "180.0")
+                    .insert("type", "soft"))
+                .insert("use", "required"));
+}
+
+void CameraFactory::add_clipping_metadata(DictionaryArray& metadata)
+{
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "near_z")
+            .insert("label", "Near Z")
+            .insert("type", "text")
+            .insert("use", "optional")
+            .insert("default", "-0.001"));
+}
+
+void CameraFactory::add_shift_metadata(DictionaryArray& metadata)
+{
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "shift_x")
+            .insert("label", "Shift X")
+            .insert("type", "text")
```

Comment:
Must be numeric.

Reply:
I tried. Numerical inputs require a min value. They crash if one is not present (a bug?).
The valid range of shifts is -inf to +inf and the values depend on the camera film sizes.
I don't have a way to assign min / max values to shifts.

Reply:
It's not a bug. You must assign min and max values, but they can be soft limits if you mark them as so. The idea is to provide a reasonable starting range for the slider to work correctly; the user is still able to go outside that range and the slider will adapt. Check how it's done elsewhere.

Reply:
Ok, but I still don't have meaninful values for min and max, soft or hard.
It is probably the same reason focal length and film width and heights are also text inputs?

Reply:
Shifting is done in multiples of the film dimensions no? In that case it's unit-less and relatively consistent across scenes, so it should be possible to infer soft min and max values.

Eventually we must turn focal length and film width/height into numeric inputs as well. Ranges aren't hard to come up with, i.e. 1-100 mm (0.001-0.100 m) for film width/height, etc.

Reply:
No. Shifting is done in meters (or the units you use for the film dimensions).

I checked what the apps we support do. Absolute shift is the result you get from camera callibration software and 3d tracking apps, camera databases and what Maya uses. 
Relative shift is used in Max and Blender. In the end I decided to use absolute for our camera.

Reply:
Ok, in that case it's the same question as the other absolute values like film dimensions, focal length, etc. Maybe we should convert them all to numeric in one go, in a subsequent PR.